builder/builder.go

```diff
@@ -8,6 +8,7 @@ import (
 	"github.com/NethermindEth/juno/core/felt"
 	"github.com/NethermindEth/juno/db"
 	"github.com/NethermindEth/juno/mempool"
+	"github.com/NethermindEth/juno/rpc"
```

Comment:
I am not sure if this is the most clean way of doing this. `builder` depending on `rpc` is code smell IMO.

Initially, to avoid declaring vm.Trace and rpc.Trace, we decided to return raw JSON from `vm.Execute()`. But we ended up having to declare `rpc.Trace` already. So I believe we need to bite the bullet and duplicate trace definition in `vm` package (or move rpc.Trace to vm) and return a properly typed `Trace` from `vm.Execute()`.